File: data-structures/list/problems/go/leetcode-283.go
package problems

func moveZeroes(nums []int) {
	p := 0
	for i := range nums {
		left := nums[p]
		right := nums[i]

		if right != 0 {
			nums[p] = right
			nums[i] = left
			p++
		}
	}
}

File: data-structures/list/problems/java/leetcode-283.java
class Solution {
    public void moveZeroes(int[] nums) {
        int p = 0;

        for (int i = 0; i < nums.length; i++) {
            int left = nums[p];
            int right = nums[i];

            if (right != 0) {
                nums[p] = right;
                nums[i] = left;
            }
        }
    }
}